src/BloomBrowserUI/pageChooser/js/page-chooser.ts

```diff
@@ -0,0 +1,156 @@
+                alert("received AddSelectedPage message before PageChooser was created");
+            return;
+
+        case 'Data':
+            logic = new PageChooser(params[1]);
+            logic.LoadInstalledCollections();
+            return;
+
+        default:
+    }
+}
+
+// this version of the test string may be useful later testing more than one template collection
+//var JSONTestString = "[{ \"templateBookUrl\": \"../../../DistFiles/factoryCollections/Templates/Basic Book/Basic Book.htm\" }, { \"templateBookUrl\": \"../../../DistFiles/factoryCollections/Templates/Basic Book/Basic Book.htm\" }]";
+// no longer using test string, but let's keep it around as documentation of what PageChooser's ctor is expecting
+//var JSONTestString = "[{ \"templateBookUrl\": \"bloom/localhost/C$/BloomDesktop/DistFiles/factoryCollections/Templates/Basic Book/Basic Book.htm\" }]";
+
+class PageChooser {
+
+    private _templateBookUrls : string;
+    private _selectedTemplatePage : JQuery;
+
+    constructor(templateBookUrls: string) {
+        if(templateBookUrls) {
+            this._templateBookUrls = templateBookUrls;
+        } else {
+            console.log('Expected url in PageChooser ctor!');
+        }
+
+        this._selectedTemplatePage = undefined;
+    }
+
+    thumbnailClickHandler( div ) : void {
+        // 'div' is an .invisibleThumbCover
+        // Mark any previously selected thumbnail as no longer selected
+        if (this._selectedTemplatePage != undefined) {
+            $(this._selectedTemplatePage).removeClass('ui-selected');
+        }
+        // Select new thumbnail
+        this._selectedTemplatePage = $( div ).parent();
+        $(this._selectedTemplatePage).addClass('ui-selected');
+        // Display large preview
+        var caption = $( '#previewCaption');
+        caption.text($('.templatePageCaption', this._selectedTemplatePage).text());
+        caption.attr('style', 'display: block;');
+        $( '#preview' ).attr( 'src', $( this._selectedTemplatePage ).find('iframe').first().attr('src'));
+        // TODO: disable the main dialog one! $( '#addPageButton' ).prop( "disabled", false );
+    } // thumbnailClickHandler
+
+    addPageClickHandler() : void {
+        if (this._selectedTemplatePage == undefined || this._templateBookUrls == undefined) {
+            return null;
+        }
+        // TODO: Add page to book here
+        var pageId = $(this._selectedTemplatePage).find('iframe').first().attr('src');
+        console.log('firing CSharp event - Selected template page: ' + pageId);
+        this.fireCSharpEvent('addPage', pageId);
+        // Mark any previously selected thumbnail as no longer selected
+        $(this._selectedTemplatePage).removeClass('ui-selected');
+    } // addPageClickHandler
+
+    LoadInstalledCollections() : void {
+        // Save html sections that will get cloned later
+        // there should only be one 'collection' at this point; a stub with one default template page
```

Comment:
Originally (now maybe YAGNI) the dialog handled more than one collection of template pages. Right now it only handles one, so the cloning of stub html is perhaps unnecessary, but I've left it in case we need it later.
(I'll put this comment in the code too.)